.github/workflows/main.yaml

```diff
@@ -24,20 +27,40 @@ jobs:
       - name: Analyze project source
         uses: invertase/github-action-dart-analyzer@v1
 
+      - uses: codecov/codecov-action@v2
+        with:
+          files: coverage/lcov.info
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        flutter-version: ['3.16.9', '3.19.6', '3.22.3', '3.24.3']
```

Comment:
Test against all supported versions of flutter, up to latest stable version. This will take some manual work to keep this up to date; I'll noodle on how to make this more dynamic as a follow-up.